core/src/main/resources/org/apache/spark/ui/static/timeline-view.js

```diff
@@ -50,9 +50,18 @@ function drawApplicationTimeline(groupArray, eventObjArray, startTime, offset) {
        return "#job-" + jobId;
       };
 
+      var getPathForJobEntry = function(baseElem) {
+        var jobIdText = $($(baseElem).find(".application-timeline-content")[0]).text();
+        var jobId = jobIdText.match("\\(Job (\\d+)\\)$")[1];
+        return "job/?id=" + jobId;
+      };
+
       $(this).click(function() {
         var jobPagePath = $(getSelectorForJobEntry(this)).find("a.name-link").attr("href");
-          window.location.href = jobPagePath
+        if (jobPagePath == undefined) {
```

Comment:
Do we need this `if` branch?
If we can unify the two pieces of code for `jobPagePath` with the new logic, can we remove away this `if` branch?

Reply:
Thanks @sarutak . we should remove it and use new logic, updated.

> @zhli1142015 Thanks for the fix. The same issue can be in `JobPage`. Could you confirm it?

Good catch, this also happens for job page. i fixed it in this PR also.
![stageJPG](https://user-images.githubusercontent.com/10524738/93276612-9a3c2280-f7f2-11ea-9c5c-f44e6ebeb6c8.JPG)